src/domain/models/errors/Apollo.ts

```diff
@@ -1,86 +1,105 @@
-import { SupportedHashingAlg } from "../../utils/hash";
-import { StorableKey } from "../keyManagement";
+import { SDKError } from "./Common";
+import { KeyTypes, StorableKey } from "../keyManagement";
+import { asArray } from "../../../utils";
+import { Curve } from "../KeyCurve";
 
-export class InvalidMnemonicWord extends Error {
-  constructor(message?: string) {
-    super(message || "Invalid mnemonic word");
+/**
+ * thrown when at least one Mnemonic word does not match in the wordlist
+ */
+export class MnemonicWordError extends SDKError {
+  constructor(words?: string[]) {
+    super(11, `Invalid mnemonic word(s) [${asArray(words).join(", ")}]`);
   }
 }
 
-export class CouldNotParseMessageString extends Error {
-  constructor(message?: string) {
-    super(message || "Could not parse message string");
-  }
+/**
+ * thrown when the number of Mnemonic words is different to expected
+ */
+export class MnemonicLengthError extends SDKError {
+  constructor() { super(112, "Mnemonics must be 12 or 24 words in length"); }
 }
 
-export class InvalidPrivateKey extends Error {
-  constructor(message?: string) {
-    super(message || "Invalid private key");
-  }
+/**
+ * // ??
+ */
+export class CouldNotParseMessageString extends SDKError {
+  constructor() { super(12, "Could not parse message string"); }
 }
 
-export class InvalidHashingAlgorithm extends Error {
-  constructor(message?: string) {
-    const supported = Object.keys(Object.fromEntries(Object.entries(SupportedHashingAlg)))
-    super(message || `Invalid Hashing Algorithm, supported ${supported.join(", ")}`);
-  }
-}
+/**
+ * thrown when restoring a key from a JWK
+ */
+// export class InvalidJWK extends SDKError {
+//   constructor() { super(13, "Invalid JWK") }
+// }
 
+/**
+ * not thrown
+ * swift throws in X25519PublicKey.verify()
+ */
+// export class KeyAgreementDoesNotSupportVerification extends SDKError {
+//   constructor() { super(14, "Invalid JWK") }
+// }
```

Comment:
Do we need to keep these commented out classes?

Reply:
we don't "need" to, it's here as it's a difference between the SDKs that we need to resolve somehow, and this is the easiest method to keep record